### DISCUSSION TITLE
ArcadeDB engine gracefully handles null LINK values

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#1088] [Show and tell]
If you have a property of type LINK, which most of the times points to a real record in another Type but sometimes it doesn't, 
you can safely insert null into a Type.LINK property and consistently use the dot notation to retrieve the linked record.
The engine is smart enough to return null instead of throwing NPE.
I think this is a neat feature: I was worried I had to handle the null case myself in the client code, but I've found out the engine does it for me, so I don't need to catch the NPE or waste storage space replacing the LINK type with an EMBEDDED type.

**Comment by lvca** [comment]
@gramian do we have this behavior in the docs already?

**Reply by gramian** [reply]
I don't think so, but I will look into it and if need be add it.
